packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -249,8 +262,13 @@ function createHook(): Hook {
   };
 }
 
-function cloneHook(hook: Hook): Hook {
+function cloneHook(hook: Hook, hookType: HookType): Hook {
```

Comment:
What about a top-level DEV-only variable like `currentHookType`. Set it to the expected type before calling `createWorkInProgressHook` and then read that in `cloneHook` and `createHook`? Then it doesn't have to be passed around and it would be more easily DCE'd for the prod build.

Reply:
I decided to read off currentHook directly!

Reply:
@threepointone with `currentHookType` you could also avoid passing `hookType` to `cloneHook` as well. I don't think GCC will [remove unused function arguments](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520SIMPLE_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F%252F%2520ADD%2520YOUR%2520CODE%2520HERE%250Afunction%2520foo(a%252C%2520unused)%2520%257B%250A%2520%2520alert('Hello%252C%2520'%2520%252B%2520name)%253B%250A%257D%250Ahello('New%2520user'%252C%2520'never%2520used')%253B%250A%250A), so the `HookType` arguments won't get removed from the prod build even though they're unused

. If you do:

```js
if (__DEV__) {
  currentHookType = RefHook
}
workInProgressHook = createWorkInProgressHook();
```

Then I believe it will get DCE'd in prod, adding no new bytes.